File: webconsole/src/main/resources/res/ui/configurationrender.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *	  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
var _tabs = false;

function pad(value) {
	if ( value < 10 ) {
		return "0" + value;
	}
	return "" + value;
}

function setSelectedTab() {
	var anchor =  window.location.hash;
	if (anchor) {
		anchor = pluginRoot + '/' + anchor.substring(1) + '.nfo';
		_tabs.find('ul > li > a').each( function(index, element) {
			if (anchor == $(element).data('href.tabs')) {
				_tabs.tabs('select', index);
				return false;
			}
			return true;
		});
	}
}

function downloadDump(ext) {
	var now = new Date();
	var name = "configuration-status-" + now.getUTCFullYear() + pad(now.getUTCMonth() + 1) + pad(now.getUTCDate()) + "-" + pad(now.getUTCHours()) + pad(now.getUTCMinutes()) + pad(now.getUTCSeconds()) + ".";
    location.href = location.href + "/" + name + ext;
}

$(document).ready(function() {
	$(window).bind( 'hashchange', setSelectedTab);
	var dlg = $('#waitDlg').dialog({
		modal    : true,
		autoOpen : false,
		draggable: false,
		resizable: false,
		closeOnEscape: false
	});

	_tabs = $('#tabs').tabs({ajaxOptions: {
		beforeSend : function() { dlg.dialog('open') },
		complete   : function() { dlg.dialog('close')},
	}}).tabs('paging');

	setTimeout(setSelectedTab, 1000);

	$('.downloadTxt').click(function() { downloadDump('txt')});
	$('.downloadZip').click(function() { downloadDump('zip')});
});

File: webconsole/src/main/java/org/apache/felix/webconsole/internal/misc/ConfigurationRender.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.felix.webconsole.internal.misc;


import java.io.PrintWriter;

import org.apache.felix.webconsole.WebConsoleUtil;


public class ConfigurationRender
{

    /**
     * Renders an info line - element in the framework configuration. The info line will
     * look like:
     * <pre>
     * label = value
     * </pre>
     *
     * Optionally it can be indented by a specific string.
     *
     * @param pw the writer to print to
     * @param indent indentation string
     * @param label the label data
     * @param value the data itself.
     */
    public static final void infoLine( PrintWriter pw, String indent, String label, Object value )
    {
        if ( indent != null )
        {
            pw.print( indent );
        }

        if ( label != null )
        {
            pw.print( label );
            pw.print( " = " );
        }

        pw.print( WebConsoleUtil.toString( value ) );

        pw.println();
    }
}
